packages/redux/src/raffles/selectors.ts

```diff
@@ -0,0 +1,224 @@
+import type { StoreState } from '../types/storeState.types';
+
+const getRaffleEntities = (state: StoreState) => getEntities(state, 'raffles');
+const getRaffleParticipationEntities = (state: StoreState) =>
+  getEntities(state, 'raffleParticipations');
+
+const getRaffleParticipations = createSelector(
+  [getRaffleParticipationEntities],
+  raffleParticipations => Object.values(raffleParticipations || {}),
+);
+
+/**
+ * Returns all raffles from state.
+ *
+ * @param state - Application state.
+ *
+ * @returns  Raffles entity or undefined.
+ */
+export const getRaffles = createSelector([getRaffleEntities], raffles =>
+  Object.values(raffles || {}),
+);
+
+/**
+ * Returns a specific raffle by its id.
+ *
+ * @param state   - Application state.
+ * @param raffleId - raffle id.
+ *
+ * @returns Raffle normalized or undefined if not existent.
+ */
+export const getRaffle = (state: StoreState, raffleId: Raffle['id']) =>
+  getEntityById(state, 'raffles', raffleId);
+
+/**
+ * Returns the estimation for a raffle
+ * @param state - Application state
+ * @param raffleId - Raffle id.
+ *
+ * @returns the object estimation raffle if exists.
+ */
+export const getRaffleEstimation = (
+  state: StoreState,
+  raffleId: Raffle['id'],
+) => getEntityById(state, 'raffleEstimations', raffleId);
+
+/**
+ * Return a raffle participation.
+ *
+ * @param state - Application state
+ * @param participationId - Participation id.
+ */
+export const getRaffleParticipation = (
+  state: StoreState,
+  participationId: RaffleParticipation['id'],
+) => getEntityById(state, 'raffleParticipations', participationId);
+
+/**
+ * Get a raffle participation for a specific raffle Id.
+ *
+ * @param state - Application state
+ * @param raffleId - raffle id.
+ *
+ * @returns a raffle participation if found otherwise undefined.
+ */
+export const getRaffleParticipationForRaffle = createSelector(
+  [
+    getRaffleParticipations,
+    (state: StoreState, raffleId: Raffle['id']) => raffleId,
+  ],
+  (raffleParticipations, raffleId) =>
+    raffleParticipations.find(
+      participation => participation.raffleId === raffleId,
+    ),
+);
+
+/**
+ * Return a raffle with estimation and participation details.
+ *
+ * @param state - Application state.
+ * @param raffleId - Raffle id.
+ *
+ */
+export const getDetailedRaffle = createSelector(
+  [getRaffle, getRaffleEstimation, getRaffleParticipationForRaffle],
+  (raffle, estimation, participation) => {
+    if (!raffle) {
+      return;
+    }
+    return {
+      ...raffle,
+      estimation,
+      participation,
+    };
+  },
+);
+
+/**
+ * Return all raffles loading from state.
+ *
+ * @param state - Application state.
+ *
+ */
+export const getIsLoadingAllRaffles = (state: StoreState) =>
+  getAllRaffles(state.raffles as RafflesState).isLoading;
+
+/**
+ * Return error raffles status from state.
+ *
+ * @param state - Application state.
+ *
+ */
+export const getErrorAllRaffles = (state: StoreState) =>
```

Comment:
Again, to be consistent with the other selectors:

```suggestion
export const getAllRafflesError = (state: StoreState) =>
```

Reply:
done